service/controller/clusterapi/v28/adapter/guest_auto_scaling_group.go

```diff
@@ -64,7 +64,12 @@ func (a *GuestAutoScalingGroupAdapter) Adapt(cfg Config) error {
 	a.HealthCheckGracePeriod = gracePeriodSeconds
 	a.RollingUpdatePauseTime = rollingUpdatePauseTime
 
-	for i, az := range key.StatusAvailabilityZones(cfg.MachineDeployment) {
+	azs, err := key.StatusAvailabilityZones(cfg.MachineDeployment)
+	if err != nil {
```

Comment:
Here is the additional error handling for refactored `StatusAvailabilityZones()`.

Reply:
I do not see why this is necessary but when it goes away later anyway, let's go.

Reply:
How is the failure behaviour then different with `WorkerAvailabilityZones`?

Reply:
`WorkerAvailabilityZones()` only returns `[]string` from provider extension spec. That cannot fail as long as spec is valid as such. As I mentioned somewhere, this is a bit confusing/controversial, but should go away once we extract node pool CF stack out.

`StatusAvailabilityZones()` is programmatic replacement for that missing provider extension in the status as mentioned in `apiextensions` PR. When we don't have status, we need to compute few things there to emulate situation that we had such thing. When the execution hasn't reached the point where required input variables exist, the computation fails and it must return error. That's why the change and difference here.

Reply:
I see. One operates on the spec. Got it.